packages/@react-aria/dnd/src/useDrag.ts

```diff
@@ -137,9 +139,20 @@ export function useDrag(options: DragOptions): DragResult {
         let rect = e.currentTarget.getBoundingClientRect();
         let x = e.clientX - rect.x;
         let y = e.clientY - rect.y;
-        if (x > size.width || y > size.height) {
+
+        if (typeof options.previewOffset === 'function') {
+          let offset = options.previewOffset(size);
+          x = offset.x;
+          y = offset.y;
+        } else if (
+          options.previewOffset === 'center' || 
+          (!options.previewOffset && (x > size.width || y > size.height))
+        ) {
           x = size.width / 2;
           y = size.height / 2;
+        } else if (options.previewOffset) {
+          x = options.previewOffset.x;
+          y = options.previewOffset.y;
```

Comment:
When testing this locally I noticed that if you set the offset larger number the drag preview will initially render with the proper offset but then slide itself to still be intersecting with the mouse. This seems to be matching native behavior as per this modified example from MDN: https://codepen.io/dl1644/pen/LYqaWYV but still feels pretty weird. 

Perhaps we should clamp the provided x,y value so the element is always intersecting the pointer?